aten/src/ATen/NamedTensorUtils.h

```diff
@@ -43,26 +43,97 @@ unify_from_right(DimnameList names, DimnameList other, const char* action = "bro
 
 namespace namedinference {
 
-// Names get propagated via the following rules:
-// 1) If result does not have names, then `names` get propagated.
-// 2) If result has names, then `names` must be equal to result.names
-void propagate_names(Tensor& result, optional<DimnameList> names);
-void propagate_names(Tensor& result, std::vector<Dimname>&& names, bool validate_names);
-CAFFE2_API void propagate_names(Tensor& result, optional<std::vector<Dimname>>&& maybe_names, bool validate_names);
-void propagate_names(TensorImpl* result, optional<DimnameList> names);
-void propagate_names(TensorImpl* result, std::vector<Dimname>&& names, bool validate_names);
-void propagate_names(TensorImpl* result, optional<std::vector<Dimname>>&& maybe_names, bool validate_names);
+// [NOTE] Writing name inference rules
+//
+// Operators that support named tensors generally look like the following:
```

Comment:
I assume that in some cases ops that just call other ops could support named tensors without doing anything?

Reply:
Yes. I'll include that in